### ISSUE TITLE
Launch webpack instead of metro during development

### ISSUE THREAD

**Issue by rahulsmehta** [callstack/repack#93]
## Describe the feature

After creating a fresh RN project (RN 0.65.1, TypeScript) & installing repack, running a build in Xcode will still start the Metro packager (instead of Webpack). I solved this by patching `packager.sh` in `react-native/scripts` to override the packager command (instead of `react-native start`, setting `PACKAGER_CMD=webpack-start` will launch webpack instead), and modifying the "Start Packager" build phase in Xcode as follows:

`react-native` patch: https://github.com/rahulsmehta/repack-example/blob/master/patches/react-native%2B0.65.1.patch

```bash
export RCT_METRO_PORT="${RCT_METRO_PORT:=8081}"
+ export PACKAGER_CMD="webpack-start"

echo "export RCT_METRO_PORT=${RCT_METRO_PORT}" > "${SRCROOT}/../node_modules/react-native/scripts/.packager.env"
if [ -z "${RCT_NO_LAUNCH_PACKAGER+xxx}" ] ; then
  if nc -w 5 -z localhost ${RCT_METRO_PORT} ; then
    if ! curl -s "http://localhost:${RCT_METRO_PORT}/status" | grep -q "packager-status:running" ; then
      echo "Port ${RCT_METRO_PORT} already in use, packager is either not running or not running correctly"
      exit 2
    fi
  else
    open "$SRCROOT/../node_modules/react-native/scripts/launchPackager.command" || echo "Can't start packager automatically"
  fi
fi

```

(note: this could also be due to error on my part while setting up my example project, so would love any feedback/feel free to close if that's the case)

## Motivation

In the interest of making setup easier, it'd be possible to pull the modified versions of `launchPackager.command`/`packager.sh` into a `scripts/` dir in repack & adding the overrides to the "Start Packager" build phase in the docs. Would there be any interest in accepting a PR for this enhancement?

**Comment by zamotany** [comment]
Yeah, if you could create a PR for that, I'm happy to merge it. My main concern is that we would have to maintain those scripts, but is should be rather straightforward to do.

**Comment by tom-sherman** [comment]
Another solution could be a note in the docs to recommend use of the `--no-packager` flag when running android/ios?

**Comment by RafikiTiki** [comment]
@rahulsmehta are you still willing to add a section in the docs regarding this configuration change? It will make make everyone's life easier at least until we have an automated way to generate a repack-configured app (like a rn-cli template). Evan after that happens it will be beneficial to have this section for all the people who are migrating brownfield apps from metro to repack.

**Comment by gronxb** [comment]
Is there any further support for this anymore ?

The metro runs when I do `react-native run-ios`, and the metro runs when I build from xcode.

Should I always develop with the following command only ?
`react-native run-ios --no-packager && react-native webpack-start`

**Comment by tom-sherman** [comment]
> The metro runs when I do react-native run-ios, and the metro runs when I build from xcode.

You can remove the build phase to stop this from happening. Mine was called "Start Packager".

**Comment by gronxb** [comment]
@tom-sherman 

I've checked the "Start Packager" and if I clear it, I won't be able to launch dev apps from xcode.
The feature to launch dev apps from xcode is exactly what I need.
I want the "Start Packager" to be webpack-start as normal.
Thanks for the feedback!

**Comment by tom-sherman** [comment]
We run the `react-native start-webpack` command manually. You can always replace that build phase with one that runs `start-webpack` instead.

**Comment by jbroma** [comment]
reopening this as this is still relevant

**Comment by jbroma** [comment]
I see two ways we can move forward with this:

1. Considering we now have `@callstack/repack-init` it might be feasible to edit project configuration files for `iOS` and `android` and replace `metro` with equivalents from `Re.Pack` there.
2. Implementing #487 might resolve this issue on it's own, with no further changes in the configuration needed

**Comment by jbroma** [comment]
From Re.Pack 4.0 onwards, the `start` command is overridden so it should start Re.Pack dev-server instead of Metro.

Note that this only works for RN  >=0.74 (RN-CLI >= 13).

Closed by #537
